File: Lab2.3/Lab2.3/TextContainer.cpp
#include "TextContainer.h"

void TextContainer::addRow(const TextRow& row) {
    rows.push_back(row);
}

void TextContainer::removeRow(int index) {
    if (index >= 0 && index < rows.size())
        rows.erase(rows.begin() + index);
}

void TextContainer::replaceRow(int index, const TextRow& row) {
    if (index >= 0 && index < rows.size())
        rows[index] = row;
}

void TextContainer::clear() {
    rows.clear();
}

int TextContainer::getRowCount() const {
    return rows.size();
}

int TextContainer::totalDigits() const {
    int total = 0;
    for (const auto& row : rows)
        total += row.countDigits();
    return total;
}

TextRow TextContainer::getRow(int index) const {
    return (index >= 0 && index < rows.size()) ? rows[index] : TextRow();
}

File: Lab2.3/Lab2.3.1/TextContainer.cs
using System.Collections.Generic;

namespace Lab5
{
    public class TextContainer
    {
        private List<TextRow> rows = new List<TextRow>();

        public void AddRow(TextRow row) => rows.Add(row);
        public void RemoveRow(int index)
        {
            if (index >= 0 && index < rows.Count)
                rows.RemoveAt(index);
        }

        public void ReplaceRow(int index, TextRow row)
        {
            if (index >= 0 && index < rows.Count)
                rows[index] = row;
        }

        public void Clear() => rows.Clear();
        public int GetRowCount() => rows.Count;

        public int TotalDigits()
        {
            int total = 0;
            foreach (var row in rows)
                total += row.CountDigits();
            return total;
        }

        public TextRow GetRow(int index)
        {
            return (index >= 0 && index < rows.Count) ? rows[index] : new TextRow();
        }
    }
}
